src/design/DesignBuilder.ts

```diff
@@ -237,6 +250,18 @@ export default class DesignBuilder {
             properties.extendedData = {
                 customStrings: [customString],
             };
+
+            if (extendedData && extendedData.length) {
+                for (const data of extendedData) {
+                    const { name, value } = data;
+                    if (name && value) {
+                        properties.extendedData = {
```

Comment:
Probably ok to just modify/mutate the properties value here since it is declared within this function. No need to spread and create a new object every time.